### PR TITLE
Allow installation alongside php-mf2 v0.5

### PR DESCRIPTION
**Pull request by barnabywalters**
Added img+alt parsing logic for compatibility with php-mf2 0.5

* Added `isImgAlt` static method to `Formats\Mf2`
* Used it in `collectArrayURLValues` and `getPlaintext` to handle cases where img+alt structures are encountered 

Additionally disabled reporting of deprecation warnings while testing for a cleaner output. All tests are passing.

### CODE REVIEW

**Comment by barnabywalters** [comment]
Additionally: [Added support for default options to be passed to the constructor](https://github.com/aaronpk/XRay/pull/111/commits/718338c90f8220c63a89683a6d2e13f510b00e54)
[718338c](https://github.com/aaronpk/XRay/pull/111/commits/718338c90f8220c63a89683a6d2e13f510b00e54)

* Default options are applied to calls to parse(), rels(), feeds() and process()
* Default options are overridden by options passed to the respective function call on a
  property-by-property basis
* Documented in README with example
* Added basic test (only for parse() behaviour)
* Also fixed a bug causing an error processing h-cards with no URL property

File: lib/XRay/Formats/Mf2.php [hunk 1]
```diff
@@ -697,18 +705,22 @@ private static function parseAsHCard($item, $http, $url, $authorURL=false) {
       if($p == 'url' && $authorURL) {
         // If there is a matching author URL, use that one
         $found = false;
-        foreach($item['properties']['url'] as $url) {
-          if(self::isURL($url)) {
-            $url = \p3k\XRay\normalize_url($url);
-            if($url == \p3k\XRay\normalize_url($authorURL)) {
-              $data['url'] = $url;
-              $found = true;
+        if (array_key_exists('url', $item['properties']) and is_array($item['properties']['url'])) {
```
Comment (aaronpk) [inline lib/XRay/Formats/Mf2.php:708]: One thing I'm trying to be consistent about in XRay is returning the same data structure regardless of the data found. So in this case, if there is no `url` property found, it should still return a `url` in the XRay response. I can't remember the normal behavior whether it should return an empty string or `false` or `null` but it should do whatever is done elsewhere.

File: tests/LibraryTest.php [hunk 2]
```diff
@@ -71,4 +71,40 @@ public function testNoHEntryMarkupWithTarget()
         $this->assertArrayNotHasKey('html', $data);
     }
 
+    public function testHandlesHCardWithoutURLProperty()
+    {
+        $url = 'http://example.com/';
+        $html = '<p class="h-card">The Mythical URLless Person</p>';
+        $xray = new p3k\XRay();
+        $data = $xray->parse($url, $html);
+        $this->assertEquals('card', $data['data']['type']);
```
Comment (aaronpk) [inline tests/LibraryTest.php:80]: This should also test to make sure the `url` property is returned and is the empty value.

**Review by aaronpk — CHANGES_REQUESTED** [review]
Thank you! Only a couple minor changes, noted inline

File: lib/XRay/Formats/Mf2.php (hunk 1 above)
Comment (barnabywalters) [reply lib/XRay/Formats/Mf2.php:708]: It turns out XRay does this already, albeit slightly inconsistently. The return value for `$data['data']` is filled at creation with `'url' => null`, and later on, if no URL property was found, it’s replaced with the value of the `$authorUrl` which was passed to `parseAsHCard`, which defaults to `false` but from my testing seems to be set to the page URL if the h-card was the main content found on the page.

So your code already makes sure that there’ll always be a `url` property, but depending on the situation it’ll either be `false` or a string, which is inconsistent with the other pre-filled properties (name and photo) which will be `null` if no matching property is found.

Comment (barnabywalters) [reply lib/XRay/Formats/Mf2.php:708]: So I don’t need to change the code I added, but and in this case I can test that the URL was set to the page URL, but in theory there’s a chance that you could end up with a weird h-card array like
```php
[
  'name' => null,
  'photo' => null,
  'url' => false
]
```

and I suppose whether or not that’s a problem is for another issue.

Comment (aaronpk) [reply lib/XRay/Formats/Mf2.php:708]: oh! well cool, that works. annoying that it's false/null but that's a different problem.
